src/gather/api.py

```diff
@@ -87,9 +110,11 @@ def collect(self):
         def ignore_import_error(_unused):
             if not issubclass(sys.exc_info()[0], ImportError):
                 raise # pragma: no cover
-        scanner = venusian.Scanner(registry=registry, tag=self)
+        scanner = venusian.Scanner(registry=registry, tag=self, strategy=strategy)
         for module in _get_modules():
             scanner.scan(module, onerror=ignore_import_error)
+        if hasattr(scanner, 'please_raise'):
```

Comment:
`hasattr` suppresses all exceptions on Python 2.7 so it's [no good](https://hynek.me/articles/hasattr/).

Why not 

```python
please_raise = getattr(scanner, 'please_raise', None)
if please_raise is not None:
    raise please_raise
```